stacks-signer/src/signer.rs

```diff
@@ -1230,8 +1226,8 @@ impl Signer {
             }
             return Ok(());
         };
-        let coordinator_id = self.get_coordinator(current_reward_cycle).0;
-        if Some(self.signer_id) == coordinator_id && self.state == State::Idle {
+        let coordinator_id = self.coordinator_selector.get_coordinator().0;
+        if self.signer_id == coordinator_id && self.state == State::Idle {
```

Comment:
I think it'd be better to update the `self.get_coordinator()` method to take an argument like `consider_miner: bool`. It's not obvious from just looking at the code that using the `coordinator_selector` field directly does this and `self.get_coordinator()` doesn't. Also, coordinator selection may be a place where the signer is making more complex decisions in the future, so I think it makes sense to just abstract the signers' invocation of `coordinator_selector` entirely within `self.get_coordinator()`

Reply:
I think if you want to use the self.get_coordinator function I would just make current_reward_cycle an optional though am not sure that makes it much clearer...

Reply:
See https://github.com/stacks-network/stacks-core/pull/4583#discussion_r1541541907

I am starting to question whether I should close this ticket without merging.  It is not gonna work as is. A signer can trigger the DKG round, but when we receive incoming packets we always get the miner id when in the middle of an active reward cycle. this introduces unnecessary complexity around unwrapping the packet and checking to see if its a DKG packet or not before either trying to treat the miner as the coordinator or the signer. I can certainly do this, but its only for the case where in epoch 2.5 we fail to run dkg before the next reward cycle which if this happened in epoch 3.0 would cause a whole host of other issues. 


@jcnelson @kantai @hstove any opinions?

Reply:
See https://github.com/stacks-network/stacks-core/pull/4598 for my proposed alternative.

Reply:
I think that the thing to do here is to match the packet type as like `MinerCoordinated` or `SignerCoordinated`, then use the miner for one and the coordinator selection from the signer set for the other. I would try not to worry _too_ much about the fact that aggregate key voting isn't possible after the reward cycle started, because follow on work may try to address that in Epoch 3.0 by allowing late aggregate key votes to go through.

Reply:
> I think maybe a better solution is to fix the error to be "is DKG not set and we are in the active reward cycle? Just immediately return as we cannot fix this"

My understanding was that we updated signers-voting to allow voting on an aggregate key even if the cycle has already started? If so, this PR seems like the right fix, because it'll execute DKG for the correct reward cycle?

Reply:
Alright, I have enabled DKG in the active reward cycle. Additionally, i have maintained the functionality that a signer could trigger a sign round (mostly for testing purposes) if no miner pubkey is known (will also help for when we make signers sign things other than blocks).